src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpMtlsHttpClientFactory.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET8_0_OR_GREATER
+
+using System.Security.Cryptography.X509Certificates;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation;
+
+/// <summary>
+/// Factory for creating HttpClient instances configured with mTLS settings.
+/// </summary>
+internal static class OtlpMtlsHttpClientFactory
+{
+    /// <summary>
+    /// Creates an HttpClient configured with mTLS settings.
+    /// </summary>
+    /// <param name="mtlsOptions">The mTLS configuration options.</param>
+    /// <param name="configureClient">Optional action to configure the client.</param>
+    /// <returns>An HttpClient configured for mTLS.</returns>
+    public static HttpClient CreateMtlsHttpClient(
+        OtlpMtlsOptions mtlsOptions,
+        Action<HttpClient>? configureClient = null)
+    {
+        ArgumentNullException.ThrowIfNull(mtlsOptions);
+
+        if (!mtlsOptions.IsEnabled)
+        {
+            var client = new HttpClient();
+            configureClient?.Invoke(client);
+            return client;
+        }
+
+        HttpClientHandler? handler = null;
+        X509Certificate2? caCertificate = null;
+        X509Certificate2? clientCertificate = null;
+
+        try
+        {
+            // Load certificates
+            if (!string.IsNullOrEmpty(mtlsOptions.CaCertificatePath))
+            {
+                caCertificate = OtlpMtlsCertificateManager.LoadCaCertificate(
+                    mtlsOptions.CaCertificatePath);
+
+                if (mtlsOptions.EnableCertificateChainValidation)
+                {
+                    OtlpMtlsCertificateManager.ValidateCertificateChain(
+                        caCertificate,
+                        OtlpMtlsCertificateManager.CaCertificateType);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(mtlsOptions.ClientCertificatePath))
+            {
+                if (string.IsNullOrEmpty(mtlsOptions.ClientKeyPath))
+                {
+                    // Check if certificate file exists to provide appropriate error message
+                    if (!File.Exists(mtlsOptions.ClientCertificatePath))
+                    {
+                        throw new FileNotFoundException($"Certificate file not found at path: {mtlsOptions.ClientCertificatePath}", mtlsOptions.ClientCertificatePath);
+                    }
+                }
```

Comment:
Is there some code missing here? If `ClientCertificatePath` is non-null and exists, `ClientKeyPath` is not provided, we don't seem to load it to use.

Reply:
Oh.. good catch!